modules/recon/profiles-contacts/bing_linkedin_contacts.py

```diff
@@ -0,0 +1,49 @@
+from recon.core.module import BaseModule
+from recon.mixins.search import BingAPIMixin
+
+
+class Module(BaseModule, BingAPIMixin):
+
+    meta = {
+        "name": "Bing LinkedIn Profile Contact Harvester",
+        "author": "Cam Barts (@cam-barts)",
+        "version": "1.0",
+        "description": "Harvests Basic Contact Information from Bing based on LinkedIn profiles",
+        "required_keys": ["bing_api"],
+        "comments": (
+            'Use Bing\'s top search results for LinkedIn urls to gather names, titles, and companies',
+            'This works for profiles that are set to public on LinkedIn'
+        ),
+        "query": "SELECT DISTINCT url FROM profiles WHERE resource='LinkedIn'",
+        "options": (),
+    }
+
+    def module_run(self, urls):
+        for url in urls:
+            self.get_contact_info(url)
+
+    def get_contact_info(self, url):
+        search_result = self.search_bing_api(url, 1)
+        if search_result and search_result[0]["url"] == url:
+            search_result = search_result[0]
+            link_title = search_result["name"]
+            name_and_title = link_title.split("|")[0]
+            name_title_company_list = name_and_title.split("-")
+            name = name_title_company_list[0].strip().split()
+            if "linkedin" not in name_title_company_list[-1].lower():
+                company = name_title_company_list[-1]
+            else:
+                company = False
+            if not company:
+                title = name_title_company_list[1]
+            else:
+                title = f"{name_title_company_list[1]} at {company}"
+            if len(name) > 2:
+                self.insert_contacts(
+                    first_name=name[0],
```

Comment:
Did you know this existed as a utility withinin the framework? Might be a better option for parsing names.

https://github.com/lanmaster53/recon-ng/blob/8781966c1559ff4fe66ad96eb508ba253ba345f5/recon/utils/parsers.py#L14

Reply:
I did notice that, but I seem to remember not liking how it was formatting out my names, and I can't remember exactly why. I will look into it again and follow up.

Reply:
I changed it to use the internal parse tool. The problem I ran into was that the parse tool doesn't know how to filter out designations like PMP and DTM. Thats a future issue to fix.